src/sentry/static/sentry/app/views/eventsV2/landing.tsx

```diff
@@ -128,7 +128,12 @@ class DiscoverLanding extends AsyncComponent<Props, State> {
 
     const cursor = decodeScalar(location.query.cursor);
     let perPage = 9;
-    if (!cursor && shouldRenderPrebuilt()) {
+
+    const canRenderPrebuilt = this.state
+      ? this.state.renderPrebuilt
```

Comment:
Maybe I'm missing something, but should this be in the type as 
```js
renderPrebuilt?: boolean
```
Then do something like:
```js
const canRenderPrebuilt = this.state.renderPrebuilt ?? shouldRenderPrebuilt();
```

So as soon as it's toggled it is no longer undefined, but before that it falls back on the persistence from local storage?

Reply:
`this.state` doesn't exist the first time this function is called, so in those cases I'm using `shouldRenderPrebuilt` to pull from localstore, then subsequent times it changes use the state since it exists now

Reply:
Oh I missed that this is called during the state for async, I see, nvm